Solutions/Marain.Tenancy.Deployment/Marain-PreDeploy.ps1

```diff
@@ -12,15 +12,21 @@ use it directly.)
 # Marain.Instance expects us to define just this one function.
 Function MarainDeployment([MarainServiceDeploymentContext] $ServiceDeploymentContext) {
 
-    $app = $ServiceDeploymentContext.DefineAzureAdAppForAppService(
-        "",
-         "TenancyAppId")
+    $app = $ServiceDeploymentContext.DefineAzureAdAppForAppService()
 
     $AdminAppRoleId = "7619c293-764c-437b-9a8e-698a26250efd"
     $app.EnsureAppRolesContain(
         $AdminAppRoleId,
         "Tenancy administrator",
-        "Full control over definition of claim permissions and rule sets",
+        "Ability to create, modify, read, and remove tenants",
```

Comment:
From which base tenant? The root? Or a specific child tenant? We *will* need to determine the difference, but it requires the claims support first.

Reply:
It is simply impossible to manage this with app roles. At the AD App role level, a particular service identity is either in a role or it isn't. Anything more fine-grained must be handled within the app. I felt it was nonetheless worth making the AD-level access control as fine-grained as possible, which in practice essentially means "readers" and "full control".